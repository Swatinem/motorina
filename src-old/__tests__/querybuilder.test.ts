import { forDialect } from "./connection";
import { test1 } from "./testtable";
import { lit } from "./";

forDialect(conn => {
  it("should build a simple query", () => {
    test1.table.select(test1.columns).execute(
      conn,
      {},
      {
        inspect({ sql }) {
          expect(sql).toMatchSnapshot();
        },
      },
    );
  });
  it("should correctly escape literals", () => {
    test1.table
      .select(test1.columns)
      .filter(test1.columns.id.eq(lit(123)))
      .execute(
        conn,
        {},
        {
          inspect({ sql }) {
            expect(sql).toMatchSnapshot();
          },
        },
      );
  });
});
