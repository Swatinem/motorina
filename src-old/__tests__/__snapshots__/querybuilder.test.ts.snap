// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using mysql should build a simple query 1`] = `"SELECT \`test1\`.\`id\`, \`test1\`.\`name\` FROM \`test1\`"`;

exports[`using mysql should correctly escape literals 1`] = `"SELECT \`test1\`.\`id\`, \`test1\`.\`name\` FROM \`test1\` WHERE \`test1\`.\`id\` = 123"`;

exports[`using postgres should build a simple query 1`] = `"SELECT \\"test1\\".\\"id\\", \\"test1\\".\\"name\\" FROM \\"test1\\""`;

exports[`using postgres should correctly escape literals 1`] = `"SELECT \\"test1\\".\\"id\\", \\"test1\\".\\"name\\" FROM \\"test1\\" WHERE \\"test1\\".\\"id\\" = 123"`;
